Add try Elyra with docker section to README (#821)
elyra-ai/elyra: README.md

## Try Elyra

#### Using Binder
You can try out some of Elyra features using the [My Binder](https://mybinder.readthedocs.io/en/latest/) service.

Click on the link below to try Elyra, on a sandbox environment, without having to install anything.

[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/elyra-ai/elyra/master?urlpath=lab/tree/binder-demo)

#### Using Docker

You can also try Elyra via docker.

The command below starts a clean Elyra environment:

```

> docker run -it -p 8888:8888 elyra/elyra:1.0.0rc3 jupyter lab --debug

Reviewer: This tag doesn't exist yet.

Author: Sorry, it was on my local machine and looks like I forgot to push (pushing now)

Reviewer: I found I had to run the image as root...
```
docker run -it -p 8888:8888 -u 0  elyra/elyra:1.0.0rc3 jupyter lab --debug --allow-root
```
otherwise, I get this:
```bash
$ docker run -it -p 8888:8888  elyra/elyra:1.0.0rc3 jupyter lab --debug
[D 17:48:06.275 LabApp] Searching ['/home/jovyan/work', '/home/jovyan/.jupyter', '/opt/conda/etc/jupyter', '/usr/local/etc/jupyter', '/etc/jupyter'] for config files
[D 17:48:06.276 LabApp] Looking for jupyter_config in /etc/jupyter
[D 17:48:06.276 LabApp] Looking for jupyter_config in /usr/local/etc/jupyter
[D 17:48:06.276 LabApp] Looking for jupyter_config in /opt/conda/etc/jupyter
[D 17:48:06.277 LabApp] Looking for jupyter_config in /home/jovyan/.jupyter
[D 17:48:06.277 LabApp] Looking for jupyter_config in /home/jovyan/work
[D 17:48:06.279 LabApp] Looking for jupyter_notebook_config in /etc/jupyter
[D 17:48:06.280 LabApp] Loaded config file: /etc/jupyter/jupyter_notebook_config.py
[D 17:48:06.281 LabApp] Looking for jupyter_notebook_config in /usr/local/etc/jupyter
[D 17:48:06.281 LabApp] Looking for jupyter_notebook_config in /opt/conda/etc/jupyter
[D 17:48:06.282 LabApp] Loaded config file: /opt/conda/etc/jupyter/jupyter_notebook_config.json
[D 17:48:06.282 LabApp] Looking for jupyter_notebook_config in /home/jovyan/.jupyter
[D 17:48:06.283 LabApp] Loaded config file: /home/jovyan/.jupyter/jupyter_notebook_config.py
[D 17:48:06.283 LabApp] Looking for jupyter_notebook_config in /home/jovyan/work
Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/traitlets/traitlets.py", line 528, in get
    value = obj._trait_values[self.name]
KeyError: 'runtime_dir'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/jupyter-lab", line 8, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.7/site-packages/jupyter_core/application.py", line 268, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/traitlets/config/application.py", line 663, in launch_instance
    app.initialize(argv)
  File "</opt/conda/lib/python3.7/site-packages/decorator.py:decorator-gen-7>", line 2, in initialize
  File "/opt/conda/lib/python3.7/site-packages/traitlets/config/application.py", line 87, in catch_config_error
    return method(app, *args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/notebook/notebookapp.py", line 1766, in initialize
    self.init_configurables()
  File "/opt/conda/lib/python3.7/site-packages/notebook/notebookapp.py", line 1380, in init_configurables
    connection_dir=self.runtime_dir,
  File "/opt/conda/lib/python3.7/site-packages/traitlets/traitlets.py", line 556, in __get__
    return self.get(obj, cls)
  File "/opt/conda/lib/python3.7/site-packages/traitlets/traitlets.py", line 535, in get
    value = self._validate(obj, dynamic_default())
  File "/opt/conda/lib/python3.7/site-packages/jupyter_core/application.py", line 99, in _runtime_dir_default
    ensure_dir_exists(rd, mode=0o700)
  File "/opt/conda/lib/python3.7/site-packages/jupyter_core/utils/__init__.py", line 13, in ensure_dir_exists
    os.makedirs(path, mode=mode)
  File "/opt/conda/lib/python3.7/os.py", line 211, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/opt/conda/lib/python3.7/os.py", line 211, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/opt/conda/lib/python3.7/os.py", line 211, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/opt/conda/lib/python3.7/os.py", line 221, in makedirs
    mkdir(name, mode)
OSError: [Errno 28] No space left on device: '/home/jovyan/.local'
```

Reviewer: After checking with colleagues, this particular issue may be related to my environment somehow.

```

> To make your Notebook work directory available to your Docker container, 

Reviewer 2: Since I'm not familiar with JupyterLab locations I thought initially this was referring to some sort of JupyterLab default directory  `${HOME}/opensource/jupyter-notebooks/`. To avoid any potential confusion I would suggest rewording the content slightly or changing the directory name to make it more clear that this can be any local directory name, such as `/path/to/local/directory`:

"To make a local directory (e.g. `${HOME}/opensource/jupyter-notebooks/`) available to your Docker container,  you can use a mount command similar to the following:"

> The command below start Elyra and mount your local Notebook
> work directory into the docker image.

Reviewer: ```suggestion
To make your Notebook work directory available to your Docker container, you can use a mount command similar to the following:
```

```
docker run -it -p 8888:8888 -v ${HOME}/opensource/jupyter-notebooks/:/home/jovyan/work -w /home/jovyan/work elyra/elyra:1.0.0rc3 jupyter lab --debug
```

> The command above will produce an output similar to the one below, where you can than find the URL to be used
> to access Elyra UI in your local browser

Reviewer: ```suggestion
These should produce output similar to that below, where you can then find the URL to be used
to access Elyra in your local browser.
```
